industrial_ci/src/docker.sh

```diff
@@ -157,17 +157,20 @@ function ici_prepare_docker_image() {
 #   (None)
 
 function ici_build_default_docker_image() {
-  export DOCKER_IMAGE="industrial-ci/$UBUNTU_OS_CODE_NAME"
+  export DOCKER_IMAGE="industrial-ci/$OS_CODE_NAME"
+  ici_generate_default_dockerfile | ici_docker_build - > /dev/null
+}
 
-  ici_docker_build - <<EOF > /dev/null
-FROM ubuntu:$UBUNTU_OS_CODE_NAME
+function ici_generate_default_dockerfile() {
+  cat <<EOF
+FROM $DOCKER_BASE_IMAGE
 
 RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
 
 RUN apt-get update -qq \
-    && apt-get -qq install --no-install-recommends -y apt-utils wget ca-certificates sudo
+    && apt-get -qq install --no-install-recommends -y apt-utils gnupg wget ca-certificates sudo
```

Comment:
Since `zesty` (of `yakkety`) `gnugp` is only recommended, but not rquired.
Our code needs it for `apt-key`.